dotnet/src/Functions/Functions.OpenApi.Extensions/Extensions/CopilotAgentPluginOpenApiDocumentVisitor.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Services;
+
+namespace Microsoft.SemanticKernel.Plugins.OpenApi.Extensions;
+
+/// <summary>
+/// Copilot Agent Plugin OpenAPI document visitor that:
+///  * Normalizes the operation IDs by replacing dots with underscores. So that the operation IDs can be used as function names in semantic kernel.
+///  * Truncates the description to the maximum allowed length.
+///  * Removes properties and parameters with the same name.
+/// </summary>
+internal sealed class CopilotAgentPluginOpenApiDocumentVisitor : OpenApiVisitorBase
+{
+    private const int MaximumDescription = 1000;
+
+    public override void Visit(OpenApiOperation operation)
+    {
+        NormalizeOperationId(operation);
+
+        TruncateOperationDescription(operation);
+
+        RemoveDuplicateOperationProperties(operation);
+    }
+
+    private static void NormalizeOperationId(OpenApiOperation operation)
+    {
+        if (operation is null || operation.OperationId is null)
+        {
+            return;
+        }
+        operation.OperationId = operation.OperationId.Replace('.', '_');
+    }
+
+    private static void TruncateOperationDescription(OpenApiOperation operation)
+    {
+        if (operation.Description?.Length > MaximumDescription)
+        {
+            operation.Description = operation.Description.Substring(0, MaximumDescription);
+        }
+    }
+
+    private static void RemoveDuplicateOperationProperties(OpenApiOperation operation)
+    {
+        HashSet<string> visitedNames = [];
+
+        var index = 0;
+
+        // Lookup for duplicate parameters and remove them
+        while (index < operation.Parameters.Count)
```

Comment:
nit: Is there a reason for `while` loop instead of `for`?

Reply:
fixed